Include/Rover/TrialView.hpp

```diff
@@ -87,14 +92,19 @@ namespace Rover {
           const Sample& operator [](std::ptrdiff_t offset) const;
 
         private:
+          const GetPtr* m_get_ptr;
           std::ptrdiff_t m_offset;
-          std::function<const Sample* (std::ptrdiff_t)> m_get_ptr;
       };
 
       //! Creates a TrialView for a trial.
       template<typename Trial>
       TrialView(const Trial& t);
 
+      //! Creates a TrialView for a trial.
+      template<typename Trial, std::enable_if_t<!std::is_lvalue_reference_v<
```

Comment:
I needed this one to make rvalue Trials work. Otherwise, they would bind to const Trial& and produce UB. The is some asymmetry though as I prolong lifetime of rvalue Trials now, but destroying Trials already read by const reference will still produce UB.
Alternatively, I could disable rvalues altogether as the View class is here to read alive Trials. Let me know if it would be better.

Reply:
Unfortunately C++ does not give us a perfect solution to this problem and the problem you describe is the same issue that ```std::string_view``` has. The C++ committee debated at length on the issue about whether to add a constructor to disable or handle r-value references and their conclusion was that such a constructor solves one problem but introduces many others, the main one being that it makes using ```TrialView<T>``` as a function parameter overly restrictive. For example...

```
ListTrial<int, int, int> auto make_trial() { ... }

void process_view(const TrialView<Sample<int, int, int>>& trial) { ... }

// This is a perfectly valid use case.
process_view(make_trial());
```

With a constructor that disables r-value references, that code won't compile. With a constructor that takes ownership of the Trial, well that has a host of problems like now copying or reassigning a ```TrialView``` is no longer a straight forward operation. The conclusion of the committee, and the one I am in agreement with, is that given the limitations it is the callers responsibility to ensure that the lifetime of the source is greater than the lifetime of the view.